components/plugin_marketplace/marketplace_item/marketplace_item_plugin/marketplace_item_plugin.tsx

```diff
@@ -284,10 +295,63 @@ export default class MarketplaceItemPlugin extends React.PureComponent <Marketpl
         this.props.actions.installPlugin(this.props.id, this.props.version);
     }
 
+    getPluginStatusButton(): React.ReactNode {
+        if (!this.props.installedVersion) {
+            return null;
+        }
+
+        const pluginEnabled = this.props.pluginStatus?.state === PluginState.PLUGIN_STATE_RUNNING || this.props.pluginStatus?.state === PluginState.PLUGIN_STATE_STARTING || this.props.pluginStatus?.state === PluginState.PLUGIN_STATE_FAILED_TO_START;
+        let switchDisabled = this.props.pluginStatus?.state !== PluginState.PLUGIN_STATE_RUNNING && this.props.pluginStatus?.state !== PluginState.PLUGIN_STATE_NOT_RUNNING && this.props.pluginStatus?.state !== PluginState.PLUGIN_STATE_FAILED_TO_START;
+
+        const limit = this.props.limits?.integrations?.enabled;
+        const usage = this.props.integrationsUsage?.enabled;
+        const ignored = Constants.Integrations.FREEMIUM_USAGE_IGNORED_PLUGINS.includes(this.props.id);
+        if (!ignored && !pluginEnabled && limit && usage && usage >= limit) {
+            switchDisabled = true;
+        }
+
+        const label = pluginEnabled ? (
+            <FormattedMessage
+                id='marketplace_modal.plugin_status.enabled'
+                defaultMessage='Enabled'
+            />
+        ) : (
+            <FormattedMessage
+                id='marketplace_modal.plugin_status.disabled'
+                defaultMessage='Disabled'
+            />
+        );
+
+        return (
+            <>
+                <Toggle
+                    id={`plugin-enable-toggle-${this.props.id}`}
```

Comment:
This looks like the right size according to the design and I don't know if this dicussion has already been had elsewhere, but I wonder if we should make the toggle larger. @annelieseklein , thoughts on making the toggle larger ? The text description already does ellipsis if it overflows, so I don't expect any overflow issues if we do this.

In my opinion, it would be worth making the toggle larger because the rest of the element's surrounding space is a link to details of the plugin, and with a small toggle, the consequence isn't just that they have to click again, but that their workflow gets broken up as they are taken to a page they did not intend to visit. No strong opinions on how much bigger to make it. Some accessibility guidlines suggest 42px (https://www.w3.org/WAI/WCAG21/Understanding/target-size.html) touch targets. My main concern here is not taking lots of users to details pages they didn't want to go to.

Reply:
@neallred Along with this, I suggested [here](https://community-daily.mattermost.com/private-core/pl/5ucwczo3o7g4mf81idi54ofj1h) that we may want to remove the "Enabled/Disabled" label next to the toggle, and instead have "Enabled" as a header for the column. Then only the toggle represents the enable/disable state.

Reply:
@neallred I agree. I think that makes sense. I've made the toggles large.